### DISCUSSION TITLE
Charm Error: Matrix is not positive definite

### DISCUSSION THREAD

**Discussion by XavierCorominas** [simnibs/simnibs#423] [7.Creating head models]
Dear Developers,

I hope this message finds you well.

I am reaching out regarding a peculiar error we have encountered and i am currently struggling to resolve. We are working with mesh models using Charm on T1 human MRI scans. When running Charm on the original, unmodified T1 MRI images, everything functions as expected. However, for the purposes of our research, we are attempting to run Charm on images that have been AC-PC aligned, and with the axis origin reset to the lower-left corner.

When using these modified images as input for charm, we occasionally encounter the following error:
numpy.linalg.LinAlgError: Matrix is not positive definite.

This issue has been raised previously but remains unresolved ([LinAlgError: Matrix is not positive definite · simnibs/simnibs · Discussion #222 (github.com)](https://github.com/simnibs/simnibs/discussions/222)).

Curiously and potentially no related with the origin of the error, the error does not occur when running Charm on a Linux system, where it performs as expected. However, when executed on a Windows machine, the error consistently arises. The error does not occur when the mri is ac-pc aligned, but consistenly appears when ac-pc aligned + origin axis reset is done before charm.

Would you have any insights into why this error might be occurring?
Please see log attached:
[charm_log.txt](https://github.com/user-attachments/files/17063860/charm_log.txt)

Thank you very much for your time and assistance.
Best regards,

Xavier COROMINAS.

**Comment by oulap** [comment]
I've seen this error pop up a few times, and it's probably related to the underlying system level math libraries. The discussion you're referring to is a bit of a separate thing as there charm was applied to non-human primate data.

In any case, a few users have managed to get around the problem by installing the intel specific numpy version in their simnibs python environment and then running charm. Please see this discussion #182

**Comment by XavierCorominas** [comment]
Hello Oula, 

Thanks for the fast reply.

I went through the forum discussions you suggested, but unfortunately, nothing worked. It’s strange that regular MRI images run perfectly on my computer, but the axis-rested ones don’t.

I’ve installed SimNIBS on my WSL on the same computer, and charm works well with the same image. I’ll call charm form my Linux for now and focus on fixing the Windows issues on the meantime.

Thanks for your help!

Best regards,

Xavier COROMINAS.
